components/NavLayout/SideBar.js

```diff
@@ -1,12 +1,122 @@
+import Link from 'next/link';
+import { useState } from 'react';
+
+import { styled, useTheme } from '@mui/material/styles';
+import { openedMixin, closedMixin, DrawerHeader, AppBar, Drawer } from '../../components/basecomponents/SideBarStyles.js';
+import { Box, Container, Stack, Avatar } from '@mui/material';
+import { ListItemButton, ListItemIcon, ListItemText } from '@mui/material';
+import { Toolbar, List, CssBaseline, Typography, IconButton } from '@mui/material';
+
+import MenuIcon from '@mui/icons-material/Menu';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import ClassOutlinedIcon from '@mui/icons-material/ClassOutlined';
+import CalendarMonthOutlinedIcon from '@mui/icons-material/CalendarMonthOutlined';
+import NotificationsOutlinedIcon from '@mui/icons-material/NotificationsOutlined';
+import SearchOutlinedIcon from '@mui/icons-material/SearchOutlined';
+
+
+import defaultProfilePic from '../../utils/constants';
 import styles from '../../utils/styles/NavLayoutStyles/SideBar.module.css';
 
 export default function SideBar({ children, ...props }) {
+  const theme = useTheme();
+  const [open, setOpen] = useState(false);
+
+  const handleDrawerOpen = () => {
+    setOpen(true);
+  };
+
+  const handleDrawerClose = () => {
+    setOpen(false);
+  };
+
+  // hard coded data, eventually change sampleUser to [username]
+  const pageUrls = {
+    0: "/sampleUser/my-classes",
+    1: "/sampleUser/calendar",
+    2: "/sampleUser/notifications",
+    3: "/sampleUser/class-catalog"
+  }
+
+  const sideBarIcons = {
```

Comment:
Up to your discretion, but we can further modularize this. Could add a line of code above 
`const iconColorStyle = { color: '#FFFFFF' }` 
and for each icon component set
`<SomeIcon style={iconColorStyle} />`
This keeps our code a little more DRY.